modules/httpAPI/dashboard/index.html

```diff
@@ -1,89 +1,41 @@
 <!DOCTYPE html>
 <html lang="en">
 <head>
+    <meta charset="UTF-8">
     <title>FlapAlertedPro</title>
-    <style>
-        #prefixTable {
-            width: 100%;
-            font-family: Arial, Helvetica, sans-serif;
-            border-collapse: collapse;
-        }
-
-        #prefixTable td,
-        #prefixTable th {
-            border: 1px solid #ddd;
-            padding: 8px;
-        }
-
-        #prefixTable tr:nth-child(even) {
-            background-color: #f2f2f2;
-        }
-
-        #prefixTable tr:hover {
-            background-color: #ddd;
-        }
-
-        #prefixTable th {
-            padding-top: 12px;
-            padding-bottom: 12px;
-            text-align: left;
-            background-color: #c04c09;
-            color: white;
-        }
-
-        .headerDiv {
-            height: 16em !important;
-            display: inline-block;
-            width: 50%;
-            box-sizing: border-box;
-        }
-
-        #wrapper {
-            box-sizing: border-box;
-        }
-
-        .disabledLink {
-            color: currentColor;
-            cursor: not-allowed;
-            opacity: 0.5;
-            pointer-events: none;
-            text-decoration: none;
-        }
-    </style>
+    <link rel="stylesheet" href="assets/css/style.css" integrity="sha384-M7r2/qL5kDteUHazfjvlljZg5t9R8ISiH1wlTsIp/vkHkSTQnLyDqAj0RIYeDcUQ">
```

Comment:
- Moving the css to a separate file increases page load time (and time to First Contentful Paint)
- SRI on css makes rapid prototyping more difficult (and does not improve security aspects as the file is served locally either way)

Reply:
The former is not quite true: it does increase the load time on first load, but on repeated load the CSS file is loaded from the cache. By swapping out the CSS file, the HTML file becomes smaller, which reduces the load time.

Reply:
To the second: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/style-src#unsafe_inline_styles

Reply:
If you now say that you want to add CSP, then there is indeed a potential security improvement at avoiding style injection if used as:
`Content-Security-Policy: require-sri-for style`

This can also be done however without the proposed change and while keeping the style inline through the use of the following CSP:
`Content-Security-Policy: style-src 'sha256-xxxx....' ...`

I am still not convinced that going through this effort is worth it as there is no protection for the JavaScript in your proposal which would be the primary vector for compromising the page. 
If you can inject `<style> `tags then you can also inject `<script>` tags.
Without protecting the JavaScript with SRI somebody could just to do the following to inject styles:
`document.getElementById('test').style = "...."`

Given that CSP adoption is low, for those not using it SRI adds no additional security features as the scripts and styles are served from the same server as the HTML.

I am open to adding SRI on the JavaScript that gets calculated and added via the Makefile to the HTML via the integrity attribute. (JS can be a separate file) I am also open to adding SRI for the inline styles using a meta tag for the CSP containing the SRI of the style via the same mechanism.

Reply:
Ideally the entire CSP could be generated by the Makefile and inserted into the HTML as a <meta> tag for all scripts and styles.
If you don't want to do that for this PR that's ok, I might have a look at it in the future

Reply:
> However, if you absolutely have to use it, there are a few mechanisms that will allow them.

I still think it would be better to outsource the CSS. As far as I understand Mozilla documentation, "style-src" with SHA should only be used as a last resort.

Reply:
> as there is no protection for the JavaScript in your proposal which would be the primary vector for compromising the page.

How do you mean that? I have also added SRI for JavaScript.

Reply:
> > as there is no protection for the JavaScript in your proposal which would be the primary vector for compromising the page.
> 
> How do you mean that? I have also added SRI for JavaScript.

The inline JavaScript tag does not have an integrity attribute containing an SRI

Reply:
Oh, I overlooked that. CSS CSP/SRI and no protection of JS of course makes no sense and is stupid.

Reply:
fixed in https://github.com/Kioubit/FlapAlertedPro/pull/3/commits/2a605c3854620fcdc8bd4eef19276b20b1637a77